catalog/core/catalog-core-impl/filter-proxy/src/main/java/ddf/catalog/filter/proxy/adapter/GeotoolsFilterAdapterImpl.java

```diff
@@ -975,55 +982,56 @@ private ExpressionValues getExpressions(Expression expression1, Expression expre
             Object delegate) {
         String propertyName;
         Object literal;
-
-        if (expression1 instanceof Function && ProximityFunction.NAME.getName()
-                .equals(((Function) expression1).getFunctionName()
-                        .getName()) && expression2 instanceof Literal) {
-            return new ExpressionValues((ProximityArguments) expression1.accept(this, delegate),
-                    expression2.accept(this, delegate));
+        String functionName;
+        List<Object> functionArgs;
+
+        if (expression1 instanceof Function && expression2 instanceof Literal) {
+            functionName = ((Function) expression1).getName();
+            functionArgs = (List) expression1.accept(this, delegate);
+            literal = expression2.accept(this,
+                    ((Function) expression1).getFunctionName()
+                            .getReturn()
+                            .getType());
+            return new ExpressionValues(functionName, functionArgs, literal);
+        } else if (expression2 instanceof Function && expression1 instanceof Literal) {
+            literal = expression2.accept(this, ((Function) expression2).getFunctionName()
```

Comment:
seeing as how I messed this up twice I will break each step out to clean it up more